### ISSUE TITLE
Please make off-line documentation self-contained (preventing privacy breach)

### ISSUE THREAD

**Issue by paulgevers** [castle-engine/castle-engine#59]
During my work on the Debian package for the 6.* version of castle-game, I noticed that (at least) the body-begin.html and body-end.html contain several references to on-line source. Off-line documentation should not automatically load on-line data (links are of course fine). Could you consider changing the off-line documentation to be self contained, i.e. provide the objects needed in the source of castle-engine?

The issue isn't new, until now (version 5.2), I have stripped out the footer.html from the documentation to prevent the piwik code from entering the off-line documentation, but the current documentation is adding lots of more on-line sources.

Side note, in Debian I will patch the documentation to use the system versions of the JavaScript files if that remains in the documentation, I don't expect/need you to solve that.

Related note, I'll for Debian I'll probably need to zip the content of https://github.com/castle-engine/cge-www to properly generate the documentation. Do I need everything, or is the data that I need limited?

**Comment by michaliskambi** [comment]
You can adjust doc/pasdoc/mk_docs.sh to not use any things from cge-www or doc/pasdoc/html-parts/ . I actually expected that, because (as mentioned in one of my emails to pkg-pascal-devel) you actually *have* to remove the --html-head, --html-body-begin and --html-body-end from the mk_docs.sh for Debian anyway. They are not implemented in the last stable PasDoc release yet.

So, in Debian package, just remove these options from mk_docs.sh:

~~~~
  --html-head ../doc/pasdoc/html-parts/head.html \
  --html-body-begin ../doc/pasdoc/html-parts/body-begin.html \
  --html-body-end ../doc/pasdoc/html-parts/body-end.html \
  --css ../doc/pasdoc/html-parts/cge-pasdoc.css \
~~~~

This way, the CGE documentation generated in the Debian package will have a "generic" look, but that's very acceptable and probably easiest for both of us. It will make the generated docs fully self-contained, and you will not have to package the cge-www contents.

*I want to do it better, but it's not easy, in short:)*

- The PasDoc documentation of CGE is "entangled" with the cge-www stuff to achieve a similar look. This way https://castle-engine.sourceforge.io/apidoc/html/index.html looks like a normal part of the Castle Game Engine website. For offline version it's nice too, this way users can jump to our website when browsing the offline docs.

- I deliberately decided to not copy the related portions of cge-www to castle-engine, because it would make maintenance harder. I would need to make sure to routinely run scripts that copy files in one direction or another (between cge-www and castle-engine repos), and I would need to maintain a list of "cge-www subset" that is necessary for API docs. The "subset" includes various things (Bootstrap, CGE core PHP files like castle_engine_functions.php and kambi-php-lib/, CGE JavaScript and CSS).

As the website was undergoing heavy changes before 6.0 release, it would be a little pain to maintain such copying script, that conservatively copies only the necessary files (and I would not like to copy more).

So:

1. It was easier to just write PHP scripts in doc/pasdoc/html-parts/ that generate the necessary HTML, using the PHP code inside ../../../../cge-www.

2. And it was easier to just reference the online files, instead of carefully copying the necessary subset (engine logo, CSS, JS) to the generated offline docs. The fact that offline docs is reported to Piwik is not something I want, but (after some switching back and forth) it was just for now easiest to leave it.

Bottom line: I want to improve, but it's not trivial for me now:) The best course of action for the Debian package is just to cut off this problem, by not using the --html-xxx PasDoc at all.

**Comment by paulgevers** [comment]
> Bottom line: I want to improve, but it's not trivial for me now:) The
> best course of action for the Debian package is just to cut off this
> problem, by not using the --html-xxx PasDoc at all.

Ok, if this is what you want, I am fine with that. I'll just have a
four-line patch in the Debian package than.

**Comment by michaliskambi** [comment]
Fixed.

It is fixed both
- in unstable version (master branch on GitHub) 
- and stable (https://github.com/castle-engine/castle-engine/tree/cge-640-apidocs branch on GitHub -- this is exactly like 6.4 release, except some modifications to doc generation, used to regenerate docs on https://castle-engine.io/apidoc/html/index.html ).

Actually, the subdirectories doc/pasdoc/ in these two branches are right now exactly equal.

Note: As you have noticed in your first post in this issue, using this for Debian packaging requires some more work:

If you want to regenerate _everything_ from sources, you will need to get https://github.com/castle-engine/cge-www . Place it as a sibling to `castle-engine` repository, and then inside doc/pasdoc/html-parts/ do `make clean` and `make` to generate the new content. See the doc/pasdoc/README.txt .

Note: We do not actually need the whole cge-www repository. Only some stuff from htdocs/images/ and htdocs/castle-engine-website-base/ -- see the doc/pasdoc/html-parts/Makefile for details.  And some PHP files --- most of them are inside htdocs/castle-engine-website-base/ , but some are outside (htdocs/castle_engine_functions.php, htdocs/castle_engine_books.php and some dependencies of them). Sorry ---  I have not separated the PHP files needed for this, yet.

So, if you want to only keep a subset of cge-ww for this purpose, right now the best I can say is to get:

~~~
htdocs/images/header_icon.png
htdocs/images/patreonlogoorange_45px.png
htdocs/castle-engine-website-base/castle-engine.css
htdocs/castle-engine-website-base/castle-engine.js
htdocs/castle-engine-website-base/bootstrap # whole dir
htdocs/castle-engine-website-base/colorbox # whole dir
htdocs/castle-engine-website-base/js # whole dir
htdocs/*.php
~~~

I fully understand that you may decide that for Debian it is easier to do as it does now (forget about html-parts/ , and do not use --html-xxx options for PasDoc). That's up to you :)
